react-components/src/metadata/admin/AttributeEnumEditGrid.js

```diff
@@ -0,0 +1,269 @@
+import TableRow from "@material-ui/core/TableRow";
+import { withStyles } from "@material-ui/core/styles";
+
+import ContentRemove from "@material-ui/icons/Delete";
+import ContentEdit from "@material-ui/icons/Edit";
+
+const normalizeDefault = (fields, change, is_default) => {
+    if (is_default) {
+        fields.forEach((field, index) => {
+            let attrEnum = fields.get(index);
+            if (attrEnum.is_default) {
+                change(field, {...attrEnum, is_default: false});
+            }
+        });
+    }
+
+    return is_default;
+};
+
+class AttributeEnumEditDialog extends Component {
+    normalizeDefaultField = (is_default) => {
+        const { fields, change } = this.props;
+        return normalizeDefault(fields, change, is_default);
+    };
+
+    render() {
+        const { open, intl, field, error, onClose } = this.props;
+
+        const dialogID = build(ids.METADATA_TEMPLATE_FORM, field, ids.DIALOG);
+        const dialogTitleID = build(dialogID, ids.TITLE);
+
+        return (
+            <Dialog
+                open={open}
+                disableBackdropClick
+                disableEscapeKeyDown
+                aria-labelledby={dialogTitleID}
+            >
+                <DialogTitle id={dialogTitleID}>{getMessage("dialogTitleEditEnumValue")}</DialogTitle>
+                <DialogContent>
+                    <Field name={`${field}.value`}
+                           label={formatMessage(intl, "value")}
+                           id={build(dialogID, ids.ENUM_VALUE)}
+                           required={true}
+                           autoFocus
+                           margin="dense"
+                           component={FormTextField}
+                    />
+                    <Field name={`${field}.is_default`}
+                           label={formatMessage(intl, "enumDefaultLabel")}
+                           id={build(dialogID, ids.ENUM_VALUE_DEFAULT)}
+                           color="primary"
+                           component={FormCheckbox}
+                           normalize={this.normalizeDefaultField}
+                    />
+                </DialogContent>
+                <DialogActions>
+                    <Button id={build(dialogID, ids.BUTTONS.CLOSE)}
+                            color="primary"
+                            disabled={!!error}
+                            onClick={onClose}
+                    >
+                        {getMessage("done")}
+                    </Button>
+                </DialogActions>
+            </Dialog>
+        );
+    }
+}
+
+class AttributeEnumEditGrid extends Component {
+    constructor(props) {
+        super(props);
+
+        this.newEnumCount = 1;
+
+        this.state = {
+            selected: -1,
+            editingEnumIndex: -1,
+        };
+    }
+
+    newEnumValue = () => formatMessage(this.props.intl, "newValue", {count: this.newEnumCount++});
+
+    onAddEnum = () => {
```

Comment:
I was using arrow-functions just to speed up dev, probably from copy-paste examples, but I think we decided against arrow-functions since they were still "experimental". Plus not every class method needs to be bound to `this` (if they are not getting passed as callbacks, etc.).

Reply:
I can refactor these arrow functions to regular class methods, to match our current standard.

Reply:
I don't mind either way.  Right now it's a little harder for me to grok the arrow functions, but I think that's just because I haven't used them much.